### PR TITLE
[PROCS-4586] Enable process checks in the core agent by default

### PR DESCRIPTION
**Pull request by daniel-taf**
#### What this PR does / why we need it:

Set the default value of datadog.processAgent.runInCoreAgent to true. This will reduce the amount of containers in the default agent configuration. 

#### Special notes for your reviewer:
We [have](https://app.datadoghq.com/s/yB5yjZ/vsc-ejz-dtt) around 55 orgs currently using this config without any reported issues. Performance implications discussed [here](https://app.datadoghq.com/notebook/9700940/process-checks-in-the-core-agent-benchmarking?notebookType=rte&view=view-mode).

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`
- [ ] For Datadog Operator chart or value changes update the test baselines (run: `make update-test-baselines`)

### CODE REVIEW

**Review by kkhor-datadog — APPROVED** [review]

**Comment by levan-m** [comment]
Change overall looks good to me. Thanks for maintaining tests for the feature! 🙇 
Few items before it can be approved and merged:
* Please merge main, so python version is updated and CI passes.
* Please bump minor version and regenerated docs - still worth bumping regardless of 7.60 check. 
* Make sure all commits are signed so it can be merged. Right now they are not.

File: charts/datadog/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 apiVersion: v1
 name: datadog
-version: 3.83.1
+version: 3.84.1
```
Comment (levan-m) [inline charts/datadog/Chart.yaml]:
```suggestion
version: 3.84.0
```

File: charts/datadog/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Datadog
 
-![Version: 3.83.1](https://img.shields.io/badge/Version-3.83.1-informational?style=flat-square) ![AppVersion: 7](https://img.shields.io/badge/AppVersion-7-informational?style=flat-square)
+![Version: 3.84.1](https://img.shields.io/badge/Version-3.84.1-informational?style=flat-square) ![AppVersion: 7](https://img.shields.io/badge/AppVersion-7-informational?style=flat-square)
```
Comment (levan-m) [inline charts/datadog/README.md]:
```suggestion
![Version: 3.84.0](https://img.shields.io/badge/Version-3.84.1-informational?style=flat-square) ![AppVersion: 7](https://img.shields.io/badge/AppVersion-7-informational?style=flat-square)
```

File: charts/datadog/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Datadog changelog
 
+## 3.84.1
```
Comment (levan-m) [inline charts/datadog/CHANGELOG.md]:
```suggestion
## 3.84.0
```

**Review by levan-m — APPROVED** [review]

**Pull request merged**